controllers/cloud.redhat.com/config/config.go

```diff
@@ -0,0 +1,71 @@
+package config
+
+import (
+	"context"
+	"strconv"
+
+	crd "cloud.redhat.com/clowder/v2/apis/cloud.redhat.com/v1alpha1"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/utils"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type ConfigPopulater interface {
+	Populate(*map[string]string)
+}
+
+func (dbc *DatabaseConfig) Populate(data *map[string]string) {
+	port, _ := strconv.Atoi((*data)["port"])
+	dbc.Hostname = (*data)["hostname"]
+	dbc.Name = (*data)["name"]
+	dbc.Password = (*data)["password"]
+	dbc.PgPass = (*data)["pgPass"]
+	dbc.Port = port
+	dbc.Username = (*data)["username"]
+}
+
+func (ojs *ObjectStoreConfig) Populate(data *map[string]string) {
+	port, _ := strconv.Atoi((*data)["port"])
+	ojs.Hostname = (*data)["hostname"]
+	ojs.AccessKey = (*data)["accessKey"]
+	ojs.SecretKey = (*data)["secretKey"]
+	ojs.Port = port
+}
+
+func MakeOrGetSecret(obj ConfigPopulater, env *crd.ClowdEnvironment, client client.Client, ctx context.Context, nn types.NamespacedName, dataInit func() map[string]string) error {
```

Comment:
I'd still prefer for the method to return a `map[string]string` instead of take a `ConfigPopulater`.  Also - minor point - but I think it's best practice to make `ctx` the first argument of a function.